src/NServiceBus.Persistence.DynamoDB/DynamoDbInstaller.cs

```diff
@@ -22,20 +22,21 @@ public DynamoDbInstaller(Installer installer, IReadOnlySettings settings)
 
         public async Task Install(string identity, CancellationToken cancellationToken = default)
         {
+            TableConfiguration outboxTableConfiguration = null;
             if (settings.IsFeatureActive(typeof(OutboxStorage))
                && settings.TryGet(out OutboxPersistenceConfiguration outboxConfig)
                && outboxConfig.CreateTable)
             {
-                await installer.CreateOutboxTableIfNotExists(settings.Get<OutboxPersistenceConfiguration>(),
-                    cancellationToken).ConfigureAwait(false);
+                outboxTableConfiguration = outboxConfig.Table;
+                await installer.CreateTable(outboxTableConfiguration, cancellationToken).ConfigureAwait(false);
             }
 
             if (settings.IsFeatureActive(typeof(SagaStorage))
                 && settings.TryGet(out SagaPersistenceConfiguration sagaConfig)
-                && sagaConfig.CreateTable)
+                && sagaConfig.CreateTable
+                && outboxTableConfiguration?.TableName != sagaConfig.Table.TableName)
```

Comment:
purely an optimization to avoid an additional attempt to create a saga in case both use the same config. However, by default the saga removes the TTL attribute name (to not cause unnecessary confusion) therefore a plain record equality check wouldn't help much.